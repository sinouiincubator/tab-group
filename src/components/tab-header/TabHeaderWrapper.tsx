import styled, { css } from 'styled-components';

interface Props {
  borderless?: boolean;
}

const borderCss = css`
  border-bottom: 1px solid rgba(0, 0, 0, 0.12);
`;

const TabHeaderWrapper = styled.div<Props>`
  display: flex;
  overflow: hidden;
  flex-shrink: 0;
  font-size: 1rem;
  color: ${(props) => props.theme.palette.text.primary};
  white-space: nowrap;
  ${(props) => !props.borderless && borderCss}

  .sinoui-tab-label-container {
    display: flex;
    flex: 1;
    overflow: hidden;
    z-index: 1;
  }

  .sinoui-tab-list {
    flex: 1;
    position: relative;
    transition: transform 0.5s cubic-bezier(0.35, 0, 0.25, 1);
    display: flex;

    /* 隐藏滚动条 */
    &::-webkit-scrollbar {
      display: none; /* chrome, safari, opera */
    }
    scrollbar-width: none; /* firefox */
    -ms-overflow-style: none; /* IE 10+, edge */
    overflow: -moz-scrollbars-none; /* old firefox */

    overflow-x: scroll;
  }

  .sinoui-tab-header-extra-content {
    line-height: 48px;
  }
`;

export default TabHeaderWrapper;
