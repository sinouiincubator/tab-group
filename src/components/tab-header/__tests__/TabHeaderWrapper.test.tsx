import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import { defaultTheme } from '@sinoui/theme';
import 'jest-styled-components';
import TabHeaderWrapper from '../TabHeaderWrapper';

it('正确渲染TabHeaderWrapper', () => {
  const tree = renderer
    .create(
      <ThemeProvider theme={defaultTheme}>
        <TabHeaderWrapper />
      </ThemeProvider>,
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('不显示底部线条', () => {
  const tree = renderer
    .create(
      <ThemeProvider theme={defaultTheme}>
        <TabHeaderWrapper borderless />
      </ThemeProvider>,
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
