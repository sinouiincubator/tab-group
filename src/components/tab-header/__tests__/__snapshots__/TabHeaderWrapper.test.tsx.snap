// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`不显示底部线条 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  overflow: hidden;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  font-size: 1rem;
  color: rgba(0,0,0,0.87);
  white-space: nowrap;
}

.c0 .sinoui-tab-label-container {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  overflow: hidden;
  z-index: 1;
}

.c0 .sinoui-tab-list {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  position: relative;
  -webkit-transition: -webkit-transform 0.5s cubic-bezier(0.35,0,0.25,1);
  -webkit-transition: transform 0.5s cubic-bezier(0.35,0,0.25,1);
  transition: transform 0.5s cubic-bezier(0.35,0,0.25,1);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-scrollbar-width: none;
  -moz-scrollbar-width: none;
  -ms-scrollbar-width: none;
  scrollbar-width: none;
  -ms-overflow-style: none;
  overflow: -moz-scrollbars-none;
  overflow-x: scroll;
}

.c0 .sinoui-tab-list::-webkit-scrollbar {
  display: none;
}

.c0 .sinoui-tab-header-extra-content {
  line-height: 48px;
}

<div
  className="c0"
/>
`;

exports[`正确渲染TabHeaderWrapper 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  overflow: hidden;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  font-size: 1rem;
  color: rgba(0,0,0,0.87);
  white-space: nowrap;
  border-bottom: 1px solid rgba(0,0,0,0.12);
}

.c0 .sinoui-tab-label-container {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  overflow: hidden;
  z-index: 1;
}

.c0 .sinoui-tab-list {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  position: relative;
  -webkit-transition: -webkit-transform 0.5s cubic-bezier(0.35,0,0.25,1);
  -webkit-transition: transform 0.5s cubic-bezier(0.35,0,0.25,1);
  transition: transform 0.5s cubic-bezier(0.35,0,0.25,1);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-scrollbar-width: none;
  -moz-scrollbar-width: none;
  -ms-scrollbar-width: none;
  scrollbar-width: none;
  -ms-overflow-style: none;
  overflow: -moz-scrollbars-none;
  overflow-x: scroll;
}

.c0 .sinoui-tab-list::-webkit-scrollbar {
  display: none;
}

.c0 .sinoui-tab-header-extra-content {
  line-height: 48px;
}

<div
  className="c0"
/>
`;
